panoramisk/manager.py

```diff
@@ -200,8 +202,24 @@ def connect(self):
                 ssl=self.config['ssl']),
             loop=self.loop)
         t.add_done_callback(self.connection_made)
+
+        if run_forever:
+            self.run_forever(on_startup, on_shutdown)
         return t
 
+    def run_forever(self, on_startup, on_shutdown):
+        """Start loop forever"""
+        try:
+            if on_startup:
+                self.loop.run_until_complete(on_startup(self))
```

Comment:
you probably can use `self.loop.create_task()` instead. As I remember old asyncio versions are not able to run_forever after running run_until_complete

Reply:
How old?
https://docs.python.org/3.5/library/asyncio-eventloop.html#asyncio.AbstractEventLoop.run_until_complete